src/Components/DataHistoryExplorer/DataHistoryExplorer.tsx

```diff
@@ -39,24 +80,371 @@ const DataHistoryExplorer: React.FC<IDataHistoryExplorerProps> = (props) => {
+            });
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction
+                    .RemoveSeries;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.itemIndex]: selectedIdx
+                }
+            ]);
+            /**
+             * when a series is removed, also remove if from usedColors reference
+             * so that whenever adding a new series we can use these available colors first
+             * before picking the next color in Highcharts palette
+             */
+            usedSeriesColorsRef.current.splice(
+                usedSeriesColorsRef.current.findIndex(
+                    (c) =>
+                        c ===
+                        state.timeSeriesTwins.find(
+                            (tsTwin) => tsTwin.seriesId === seriesId
+                        ).chartProps.color
+                ),
+                1
+            );
+        },
+        [state.timeSeriesTwins, dispatch, sendDataHistoryExplorerUserTelemetry]
+    );
+
+    const handleOnViewModeChange = useCallback(
+        (viewerMode: TimeSeriesViewerMode) => {
+            dispatch({
+                type: DataHistoryExplorerActionType.SET_VIEWER_MODE,
+                payload: { viewerMode: viewerMode }
+            });
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction.ChangeView;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.view]: viewerMode
+                }
+            ]);
+        },
+        [dispatch, sendDataHistoryExplorerUserTelemetry]
+    );
+
+    const handleOnChartOptionChange = useCallback(
+        (chartOptions: ITimeSeriesCommandBarOptions) => {
+            dispatch({
+                type:
+                    DataHistoryExplorerActionType.SET_COMMAND_BAR_CHART_OPTIONS,
+                payload: { chartOptions: chartOptions }
+            });
+            logDebugConsole(
+                'debug',
+                `Chart options changed: ${JSON.stringify(chartOptions)}`
+            );
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction
+                    .ChangeChartOption;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.chartOptions]: chartOptions
+                }
+            ]);
+        },
+        [
+            state.timeSeriesTwins,
+            dispatch,
+            logDebugConsole,
+            sendDataHistoryExplorerUserTelemetry
+        ]
+    );
+
+    const handleOnDownloadTableClick = useCallback(() => {
+        downloadText(
+            JSON.stringify(
+                tableData.map((d) => {
+                    const { property, ...rest } = d;
+                    delete rest.seriesId;
```

Comment:
how robust is this? Should we instead map over the properties we know we want? I'm just wondering if it will break in the future if we add another property to the object.

Reply:
yeah just in case if we extend table columns in the future, I changed it to assign what we want